docs/theming.md

```diff
@@ -0,0 +1,299 @@
+
+## Theming with CSS custom properties
+
+MWC supports theming using [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties). 
+Custom properties are set just like standard CSS properties:
+
+```css
+html, body  {
+    --mdc-theme-primary: fuchsia;
+}
+```
+
+In addition to setting custom properties to customize the appearance of MWC components, you can also 
+use them to style your own components (or generic HTML elements):
+
+```css
+div.myCard {
+    background-color: var(--mdc-theme-primary);
+}
+```
+
+### Custom properties and older browsers
+
+CSS custom properties are supported on most modern browsers. However, if you need to support older browsers,
+you'll need to take a few extra steps. (These steps primarily apply to Internet Explorer 11.)
+
+1.  Load the web components polyfills, as described in the [Getting started guide](../README.md#supporting-older-browsers).
+
+2.  Load `CustomStyleInterface` (after loading the web components polyfills):
+
+    ```html
+    <script src="@webcomponents/shadycss/custom-style-interface.min.js"></script>
+    ```
+
+3.  Identify any document level style elements that define custom properties 
+    (for example, by adding a class): 
+
+    ```html
+    <style class="document-style">
+    html {
+      --mdc-theme-primary: brown;
+    }
+    </style>
+    ```
+
+4.  From script, call `addCustomStyle` for all document level style elements. 
+
+    ```js
+    window.ShadyCSS.CustomStyleInterface.addCustomStyle(document.querySelector('style.document-style'));
+    ```
+
+    This code assumes a single document-level style element. If you have more than one, you'll need to modify the code.
+    See []()
+
+With these steps, the polyfills provide limited support for CSS custom properties. 
+See [Limitations](https://github.com/webcomponents/polyfills/tree/master/packages/shadycss#limitations)
+in the ShadyCSS README for details.
+
+## Colors
+
+Material Design theming uses two main colors: **primary** and **secondary**. The primary color is used
+throughout most of the application and components, as the main color for your application. The secondary color is used
+for floating action buttons and other interactive elements, serving as visual contrast to the primary.
+
+In addition to the primary and secondary colors, Material Theming also defines a _background_ color for the app, 
+as well as a _surface_ color, which is used used as a background in components.
+
+Finally, Material Theming has a number of text colors, which are used for rendering text and other shapes on top of the primary,
+secondary and background colors. These are specified as either dark or light, in order to provide sufficient contrast to
+what's behind them, and have
+[different levels of opacity depending on usage](https://material.io/design/color/the-color-system.html#):
+- Primary, used for most text.
```

Comment:
This still says "primary" and "secondary" for colors--is this out of date?

https://material.io/design/color/the-color-system.html#color-usage-palettes

I know we use "high emphasis" and "medium emphasis" for contained buttons and outlined buttons, but I haven't seen it for colors.

Reply:
There are two types of color: theme color and text color.

Theme colors are primary and secondary, what you linked.

Text colors used to be called primary and secondary as well, in addition to disabled, hint, and icon. To avoid confusion with the theme colors, the first two are called high and medium emphasis.

So now, "primary" should only ever refer to the primary theme color. There'd be no confusion between "is this primary the theme or primary the text"?

Reply:
OH I see ... Yeah, I totally didn't get that. (I did wonder how "primary on background" was different from "primary," but ...